src/userobjects/ValueFractionHeuristicUserObject.C

```diff
@@ -0,0 +1,53 @@
+#include "ValueFractionHeuristicUserObject.h"
+
+registerMooseObject("CardinalApp", ValueFractionHeuristicUserObject);
+
+InputParameters
+ValueFractionHeuristicUserObject::validParams()
+{
+
+  InputParameters param = ClusteringHeuristicUserObjectBase::validParams();
+  param.addRequiredParam<Real>("upper_fraction", "upper percentage of error for the heuristics");
+  param.addRequiredParam<Real>("lower_fraction", "lower percentage of error for the heuristics");
+  param.addClassDescription(" mimics the error fraction marker in moose");
+
+  return param;
+}
+ValueFractionHeuristicUserObject::ValueFractionHeuristicUserObject(const InputParameters & params)
+  : ClusteringHeuristicUserObjectBase(params),
+    _upper_fraction(getParam<Real>("upper_fraction")),
+    _lower_fraction(getParam<Real>("lower_fraction")),
+    _max(std::numeric_limits<Real>::max()),
+    _min(std::numeric_limits<Real>::min())
```

Comment:
Aren't these initialized backwards?  How will anything ever be larger than `max` to update it?